### PR TITLE
Add accuracy logging and robust evaluation

### PR DESCRIPTION
**Pull request by blokidde**
## Summary
- improve TensorBoard logging paths in `compare_models.py`
- return validation accuracy and include in results table
- silence predictions and print progress messages
- handle training errors gracefully

## Testing
- `python3 -m py_compile compare_models.py`
- `python3 compare_models.py --help`
- `python3 compare_models.py --data_dir ./animal_photos/simple_images --img_size 224 --batch_size 1 --epochs 1` *(fails: URL fetch failure while downloading weights)*

------
https://chatgpt.com/codex/tasks/task_e_6840a77e57e08330a05da3b8506c081d

### CODE REVIEW

**Comment by blokidde** [comment]
testing codex and the way it can read the codebase, add to the codebase and use commands
